server/sonar-server/src/main/java/org/sonar/server/metric/ws/MetricConverter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.metric.ws;
+
+import com.google.common.base.Function;
+import java.util.Locale;
+import javax.annotation.Nonnull;
+import org.sonar.api.i18n.I18n;
+import org.sonar.db.metric.MetricDto;
+import org.sonarqube.ws.Common.Metric;
+
+import static com.google.common.collect.FluentIterable.from;
+import static org.sonar.server.measure.ws.MeasureValueFormatter.formatNumericalValue;
+
+public class MetricConverter {
+  private final MetricDtoToWsMetric converter;
+
+  public MetricConverter(I18n i18n) {
+    this.converter = new MetricDtoToWsMetric(i18n);
+  }
+
+  public Iterable<Metric> fromDtos(Iterable<MetricDto> metricDtos) {
+    return from(metricDtos).transform(converter).toList();
+  }
+
+  private static class MetricDtoToWsMetric implements Function<MetricDto, Metric> {
+    private final I18n i18n;
+
+    private MetricDtoToWsMetric(I18n i18n) {
+      this.i18n = i18n;
+    }
+
+    @Override
+    public Metric apply(@Nonnull MetricDto dto) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)